pkg/config/security/api_client.go

```diff
@@ -0,0 +1,335 @@
+		return nil
+	})
+
+	if err != nil {
+		vc.Logger.Errorf("unable to get the API clients; err=%s", err.Error())
+		return nil, "", err
+	}
+
+	if response.StatusCode() != http.StatusOK {
+		if err := errorsx.HandleCommonErrors(ctx, response.HTTPResponse, "unable to get API clients"); err != nil {
+			vc.Logger.Errorf("unable to get the API clients; err=%s", err.Error())
+			return nil, "", err
+		}
+
+		vc.Logger.Errorf("unable to get the API clients; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API clients")
+	}
+
+	apiclientsResponse := &APIClientListResponse{}
+	if err = json.Unmarshal(response.Body, &apiclientsResponse); err != nil {
+		vc.Logger.Errorf("unable to get the API clients; err=%s, body=%s", err, string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API clients")
+	}
+
+	return apiclientsResponse, response.HTTPResponse.Request.URL.String(), nil
+}
+
+func (c *ApiClient) UpdateAPIClient(ctx context.Context, apiClientConfig *APIClientConfig) error {
+	vc := contextx.GetVerifyContext(ctx)
+	client := openapi.NewClientWithOptions(ctx, vc.Tenant, c.Client)
+	if apiClientConfig == nil {
+		vc.Logger.Errorf("client object is nil")
+		return errorsx.G11NError("client object is nil")
+	}
+
+	id, err := c.GetAPIClientId(ctx, apiClientConfig.ClientName)
+	if err != nil {
+		vc.Logger.Errorf("unable to get the client ID for API client '%s'; err=%s", apiClientConfig.ClientName, err.Error())
+		return errorsx.G11NError("unable to get the client ID for API client '%s'; err=%s", apiClientConfig.ClientName, err.Error())
+	}
+
+	body, err := json.Marshal(apiClientConfig)
+	if err != nil {
+		vc.Logger.Errorf("unable to marshal the API client; err=%v", err)
+		return errorsx.G11NError("unable to marshal the API client; err=%v", err)
+	}
+
+	response, err := client.UpdateAPIClientWithBodyWithResponse(ctx, id, "application/json", bytes.NewBuffer(body), func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Accept", "application/json")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", vc.Token))
+		return nil
+	})
+	if err != nil {
+		if err := errorsx.HandleCommonErrors(ctx, response.HTTPResponse, "unable to update API client"); err != nil {
+			vc.Logger.Errorf("unable to update the API client; err=%s", err.Error())
+			return err
+		}
+		vc.Logger.Errorf("unable to update API client; err=%v", err)
+		return errorsx.G11NError("unable to update API client; err=%v", err)
+	}
+	if response.StatusCode() != http.StatusNoContent {
+		vc.Logger.Errorf("failed to update API client; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return errorsx.G11NError("failed to update API client ; code=%d, body=%s", response.StatusCode(), string(response.Body))
+	}
+
+	return nil
+
+}
+
+func (c *ApiClient) GetAPIClientId(ctx context.Context, clientName string) (string, error) {
```

Comment:
I suggest this should not be a public function. Instead, we should have `GetAPIClientByName` and `GetAPIClient` (which is by ID).